src/Microsoft.Restier.Core/Query/DefaultQueryHandler.cs

```diff
@@ -146,17 +151,19 @@ public override Expression Visit(Expression node)
                     }
 
                     // Inspect the visited node
-                    this.Inspect();
+                    this.InspectAsync();
```

Comment:
This can return before inspecting actually completes, and so you ignore inspection result.